MWT/how-to-container.rst

```diff
@@ -1,6 +1,66 @@
 How-To Create a Container
 =========================
 
+To create a custom container, a new class should be created, extending the ``Container`` class.
+This new class may define a constructor and setter methods in order to provide a way for the user to configure the container, such as its orientation.
+Container subclasses have to implement two methods and may override optional methods, as explained in the following sections.
+
+Implementing the mandatory methods
+----------------------------------
+
+This section explains how to implement the two mandatory methods of a container subclass.
+
+Computing the optimal size of the container
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``computeContentOptimalSize()`` method is called by the MWT framework in order to know the optimal size of the container.
+The optimal size of the container should be big enough so that each child can be allocated a size at least as big as its own optimal size.
+For example, the optimal width of a horizontal list is the sum of the optimal width of every child, and the optimal height is the maximum optimal height of all children.
+
+The container is responsible for computing the optimal size of every child. To do so, the ``computeChildOptimalSize()`` method should be called for every child.
+After this method is called, the optimal size of the child can be retrieved by calling ``getWidth()`` and ``getHeight()`` on the child widget.
+
+The ``Size`` parameter of the ``computeContentOptimalSize()`` method initially contains the size available for the container.
+An available width or height equal to ``Widget.NO_CONSTRAINT`` means that the optimal size should be computed without considering any restriction on the respective axis.
+Before the method returns, the size object should be set to the optimal size of the container.
+
+"blurry point": should the optimal size always be lower than the available size?
+
+Laying out the children of the container
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``layOutChildren()`` method is called by the MWT framework in order to lay out every child of the container.
+If a child is laid out outside the bounds of the container (partially or fully), only the part of the widget which is within the container bounds will be visible.
+
+The container is responsible for laying out each child. To do so, the ``layOutChild()`` method should be called for every child.
+Before this method is called, the optimal size of the child can be retrieved by calling ``getWidth()`` and ``getHeight()`` on the child widget.
+
+When laying out a child, its bounds have to be passed as parameter. The position will be interpreted as relative to the position of the container content.
```

Comment:
I think that the `container content` needs to be explained: the boxes (outlines and border) are removed.

Reply:
And the boxes must also be explained in the previous section (optimal size).

Reply:
i think the notion of content should be explained in architecture page, in the boxing model section

Reply:
Of course, but I think that here, we should explain when the boxes are used.

Reply:
done